### DISCUSSION TITLE
Integration with form-actions-nuxt

### DISCUSSION THREAD

**Discussion by Hebilicious** [Mini-ghost/vorms#26] [General]
Hello,
I'm currently writing some examples to explain how to integrate the [form-actions-nuxt](https://github.com/Hebilicious/form-actions-nuxt) modules with 3rd party validation/form libraries. I just discovered vorms today, and started with this example 
https://github.com/Hebilicious/form-actions-nuxt/blob/main/playgrounds/vorms-zod/pages/login.vue
It seems to me that in order to manually check the validation you have to do the following : 

```ts
    const result = await validateForm() // Validate with the library
    if (Object.keys(result).length === 0) { // Vorms returns an empty object if the form is valid.
      await submitForm() // Submit the form if valid ...
    }
``` 

Is this correct, or do you suggest another approach ? 

By the way your library looks really great ! Will definitely use it in future projects.

**Comment by Mini-ghost** [comment]
Hello,

I have read the form-actions-nuxt readme, and I am truly excited about the development of this library. It looks really impressive.

Since form-actions-nuxt relies on the native features of `<form>` for form submission, it appears that performing validation before submission can only be done manually within the run function. Therefore, I believe the approach mentioned above is perfectly acceptable.

I sincerely appreciate you taking the time to write documentation on integrating Vorms with form-actions-nuxt. If you have any suggestions or recommendations regarding the integration, I am more than open to discussing them.

Thank you for your attention and support.
